konveyor/pelorus
# Allow commit time exporter to use information from annotations.

Change to resolve #317.

The idea is taken from PR #381, however the implementation allows to have minimal required annotations to calculate commit time. It also allows to define custom annotations.


## Testing Instructions
1. Create simple python app, e.g. hello_world_app
```
#!/usr/bin/env python3
print("Hello World")
```

2. Create new BuildConfig and label it with the required `app.kubernetes.io/name`. We will use `default` namespace.
```
oc new-build python --name=python-binary-build --binary=true
oc label bc python-binary-build app.kubernetes.io/name=python-binary-build
```

3. Run the build using hello_world_app
```
oc start-build bc/python-binary-build --from-file=./hello_world_app --follow
```

4. Check the build ID
```
oc get builds -n default
```  

5. Annotate the build with the valid hash and repo URL. For test purposes it can be any public repo, however in the real world the hash and repo URL should match the source code used for the Build. --overwrite flag can be used if for test one build should be annotated more then once.
```
oc annotate build python-binary-build-1 --overwrite \
io.openshift.build.commit.id=7810f2a85d5c89cb4b17e9a3208a311af65338d8 \
io.openshift.build.source-location=http://github.com/konveyor/pelorus
```

6. Deploy pelorus with commit time exporter configured to use in our case github and check if the collected timestamp includes the app `python-binary-build`, similarly to:
```
Collected commit_timestamp{ namespace=default, app=python-binary-build, commit=286e299c5c9166ba2c554f3ee3db4592789198b5, image_sha=sha256:62dd2f7548f7d1c08b747f4ae1e1f7017a7d75767f3372a049642c700bd970e2 } 1654787319.0
```

Co-authored-by: Mike Hepburn <eformat@gmail.com>
@redhat-cop/mdt
